core/src/Cuda/Kokkos_Cuda_Task.cpp

```diff
@@ -229,20 +229,28 @@ printf("cuda_task_queue_execute before\n");
 #endif
 
   // Query the stack size, in bytes:
-  //
-  // size_t stack_size = 0 ;
-  // CUDA_SAFE_CALL( cudaDeviceGetLimit( & stack_size , cudaLimitStackSize ) );
-  //
+
+  size_t previous_stack_size = 0 ;
+  CUDA_SAFE_CALL( cudaDeviceGetLimit( & previous_stack_size , cudaLimitStackSize ) );
+
   // If not large enough then set the stack size, in bytes:
-  //
-  // CUDA_SAFE_CALL( cudaDeviceSetLimit( cudaLimitStackSize , stack_size ) );
+
+  const size_t larger_stack_size = 8192 ;
+
+  if ( previous_stack_size < larger_stack_size ) {
+    CUDA_SAFE_CALL( cudaDeviceSetLimit( cudaLimitStackSize , larger_stack_size ) );
+  }
 
   cuda_task_queue_execute<<< grid , block , shared_total , stream >>>( queue , shared_per_warp );
 
   CUDA_SAFE_CALL( cudaGetLastError() );
 
   CUDA_SAFE_CALL( cudaDeviceSynchronize() );
 
+  if ( previous_stack_size < larger_stack_size ) {
+    CUDA_SAFE_CALL( cudaDeviceSetLimit( cudaLimitStackSize , previous_stack_size ) );
+  }
+
```

Comment:
Do we need to take the previous stack size ? Besides this task-dag model (which uses relocatable device code), the stack size can be statically determined whenever parallel_for is launched (?).

Reply:
Returning the GPU settings to their original state after the task dag executes.

Reply:
Be aware that this will add a few microseconds driver call overhead

Reply:
Later, it would be nice if Kokkos allow a user to control the stack size for task-dag scheduler.